test/fn/edit-contacts.spec.js

```diff
@@ -0,0 +1,154 @@
+const { expect, assert } = require('chai');
+const rewire = require('rewire');
+const toDocs = require('../../src/fn/csv-to-docs');
+const PouchDB = require('pouchdb-core');
+PouchDB.plugin(require('pouchdb-adapter-memory'));
+const fs = require('../../src/lib/sync-fs');
+const editContactsModule = rewire('../../src/fn/edit-contacts');
+const processDocs = editContactsModule.__get__('processDocs');
+const fetchDocumentList = editContactsModule.__get__('fetchDocumentList');
+const environment = require('../../src/lib/environment');
+const sinon = require('sinon');
+
+const pouch = new PouchDB('edit-contacts', { adapter: 'memory' });
+
+// specifying directory paths to use
+const editContactsPath = `data/edit-contacts/documents`;
+const csvDir = `${editContactsPath}/csv`;
+const saveDocsDir = `${editContactsPath}/json_docs`;
+const expectedDocsDir = `${editContactsPath}/expected-json_docs`;
+const filesToUpload = fs.recurseFiles(`${editContactsPath}/server-contact_docs`).filter(name => name.endsWith('.json'));
+const saveJsonDoc = doc => fs.write(`${saveDocsDir}/${doc._id}.doc.json`, toDocs.toSafeJson(doc) + '\n');
+const countFilesInDir = path => fs.fs.readdirSync(path).length;
+const csvFiles = ['contact.csv'];
+
+
+const docs = filesToUpload
+  .map(file => {
+    const doc = fs.readJson(file);
+    return doc;
+  });
+
+const uploadDocuments = (docs) => {
+  return pouch.bulkDocs(docs);
+};
+
+const fetchDocuments = () => {
+  return pouch.allDocs({include_docs: true});
+};
+
+const processDocuments = (docType, csv, contactDocs, args) => {
+  return processDocs(docType, csv, contactDocs, args);
+};
+
+function compareDocuments(){
+  fs.recurseFiles(expectedDocsDir)
+      .map(file => fs.path.basename(file))
+      .forEach(file => {
+        const expected  = fs.readJson(`${expectedDocsDir}/${file}`);
+        const generated = fs.readJson(`${saveDocsDir}/${file}`);
+        delete generated._rev;
+        expect(generated).to.deep.eq(expected);
+      });
+}
+
+describe('edit-contacts', function() {
+
+  before(async () => {
+    await uploadDocuments(docs);
+    sinon.stub(environment, 'pathToProject').get(() => editContactsPath);
+    const pouchDb = sinon.stub();
+    pouchDb.returns(pouch);
+    editContactsModule.__set__('pouch', pouchDb);
+  });
+
+  after(async () => pouch.destroy());
```

Comment:
`before` and `after` run just once, before and after the whole suite. 
I think you should do the tear-up and tear-done for every test, using `beforeEach` and `afterEach`.
Also, make sure to clear the contents of `saveDocsDir` and empty your database!

Reply:
Done.